client/js/components/procedure/StatementSegmentsList/StatementSegment.vue

```diff
@@ -801,10 +802,12 @@ export default {
         type: 'StatementSegment',
         attributes: {
           ...this.segment.attributes,
-          customFields: {
-            ...this.segment.attributes.customFields,
-            ...payload.data.attributes?.customFields
-          }
+          ...(hasCustomFields && {
+            customFields: {
+              ...this.segment.attributes.customFields,
+              ...payload.data.attributes?.customFields
+            }
+          })
```

Comment:
just a really personal opinion, because I don't like to use && to then return something:
```suggestion
          ...(hasCustomFields ? {
            customFields: {
              ...this.segment.attributes.customFields,
              ...payload.data.attributes?.customFields
            }
          } : {})
```